### PR TITLE
Improve the reliability of the deployment test scripts

### PR DESCRIPTION
**Pull request by TimMoore**

### CODE REVIEW

**Comment by TimMoore** [comment]
I'm still seeing some flaky runs, but seems to be better overall.

File: .travis.yml [hunk 1]
```diff
@@ -73,14 +73,14 @@ jobs:
 stages:
   - test
   - name: deploy
-    if: fork = false
+    if: type = push # only deploy branches that are pushed directly to the repo
```
Comment (ignasi35) [inline .travis.yml:76]: 👍🏼

File: shopping-cart/deploy/scripts/common/deployment-tools.sh [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+    do
+        sleep 2
+        (( count = count + 1 ))
+        if [ $count -gt 10 ]
+        then
+            echo "$NAMESPACE couldn't be deleted."
+            exit 1
+        fi
+    done
+    echo "$NAMESPACE was deleted."
```
Comment (ignasi35) [inline shopping-cart/deploy/scripts/common/deployment-tools.sh:29]: nice!

File: shopping-cart/deploy/scripts/common/deployment-tools.sh [hunk 3]
```diff
@@ -18,8 +39,9 @@ useNamespace() {
 ## 1. NAMESPACE
 createNamespace() {
     NAMESPACE=$1
-    echo -n "Waiting for $NAMESPACE to be created."
-    while [ "$(oc get projects --no-headers=true | grep $NAMESPACE | wc -l)" -ne 1 ]
+    echo "Waiting for $NAMESPACE to be created."
+    local -i count=0
+    while [ -z "$(oc get project $NAMESPACE --no-headers=true)" ]
```
Comment (ignasi35) [inline shopping-cart/deploy/scripts/common/deployment-tools.sh:44]: a lot cleaner.

File: shopping-cart/deploy/scripts/common/deployment-tools.sh [hunk 4]
```diff
@@ -143,8 +165,10 @@ buildRoute() {
...
-  ## 
-  ## The hostname is set to '${SERVICE_NAME}-$NAMESPACE.$OPENSHIFT_SERVER'  since that's 
-  ## the convention used by openshift when using `oc expose service`
-  oc create route edge --service=$SERVICE_NAME --hostname=${SERVICE_NAME}-$NAMESPACE.$OPENSHIFT_SERVER  || exit 1
+  ##
+  ## The hostname is set to '${SERVICE_NAME}-$NAMESPACE.$OPENSHIFT_SERVER', since that's
+  ## the convention used by openshift when using `oc expose service`.
+  ## The hostname is truncated after 63 characters, the maximum length for a DNS hostname.
+  local hostname="${SERVICE_NAME}-$NAMESPACE"
+  oc create route edge --service=$SERVICE_NAME --hostname=${hostname:0:63}.$OPENSHIFT_SERVER  || exit 1
```
Comment (ignasi35) [inline shopping-cart/deploy/scripts/common/deployment-tools.sh:173]: TIL, max length is 64 chars.

**Review by ignasi35 — APPROVED** [review]

**Comment by ignasi35** [comment]
Did you `block-merge` for any special reason?

File: shopping-cart/deploy/scripts/common/deployment-tools.sh (hunk 4 above)
Comment (TimMoore) [reply shopping-cart/deploy/scripts/common/deployment-tools.sh:173]: [63!](https://tools.ietf.org/html/rfc1034#section-3.1) I guess they wanted it to fit into a 64-element char array, including a trailing null byte. It's only by coincidence that I choose a branch name long enough to exceed it... and it took me a little while to understand what was wrong.

**Comment by TimMoore** [comment]
> Did you `block-merge` for any special reason?

Originally, yes, but then I forgot to clear it when I finished. Done!

**Pull request merged**